### ISSUE TITLE
Port number not set

### ISSUE THREAD

**Issue by mqaa** [jjmutumi/tus_client#19]
When the upload url contains a port number that is not set correctly by the _parseUri function. I have created a pull request with the very simple fix for this issue.

**Comment by mqaa** [comment]
Not sure why but I don't see the pull request anymore... This is the change needed to make it work:

```
Uri _parseUrl(String urlStr) {
    if (urlStr.contains(",")) {
      urlStr = urlStr.substring(0, urlStr.indexOf(","));
    }
    Uri uploadUrl = Uri.parse(urlStr);
    if (uploadUrl.host.isEmpty) {
      uploadUrl = uploadUrl.replace(host: url.host, **port: url.port**);
    }
    if (uploadUrl.scheme.isEmpty) {
      uploadUrl = uploadUrl.replace(scheme: url.scheme);
    }
    return uploadUrl;
  }
```

**Comment by jjmutumi** [comment]
Thanks for the input @mqaa. I really wish you could get that pull request? Otherwise I can make the change and add test for this case a bit later.

**Comment by mqaa** [comment]
@jjmutumi I have created a pull request with the required change to make the client work with urls containing a port number.

**Comment by jjmutumi** [comment]
Fixed by #20
